src/test/java/io/wgj/ctci/StackTest.java

```diff
@@ -0,0 +1,50 @@
+package io.wgj.ctci;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
```

Comment:
Recommend adding a test that has a combination of interleaved push() and pop() calls to ensure you're tracking position correctly.

Reply:
Fixed in cd6dc83cc0daf94827e214ba2a7c26f33c9aef00